src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -305,6 +305,7 @@ mongoc_client_pool_pop (mongoc_client_pool_t *pool)
 again:
    if (!(client = (mongoc_client_t *) _mongoc_queue_pop_head (&pool->queue))) {
       if (pool->size < pool->max_pool_size) {
+         // JFW: client = _mongoc_client_new_from_uri (pool->topology);
```

Comment:
It looks like _mongoc_client_new_from_topology() is correct to me.

Reply:
Agreed.